src/AutoRest.CSharp/Mgmt/Decorator/ParentDetection.cs

```diff
@@ -86,14 +84,8 @@ private static IEnumerable<MgmtTypeProvider> FindScopeParents(ResourceTypeSegmen
 
             foreach (var type in parameterizedScopeTypes)
             {
-                if (type == ResourceTypeSegment.ManagementGroup)
-                    yield return MgmtContext.Library.ManagementGroupExtensions;
-                else if (type == ResourceTypeSegment.ResourceGroup)
-                    yield return MgmtContext.Library.ResourceGroupExtensions;
-                else if (type == ResourceTypeSegment.Subscription)
-                    yield return MgmtContext.Library.SubscriptionExtensions;
-                else if (type == ResourceTypeSegment.Tenant)
-                    yield return MgmtContext.Library.TenantExtensions;
+                if (MgmtContext.Library.TryGetExtension(type, out var extension))
```

Comment:
TODO -- we might have the case of one resource type to multiple extensions. Therefore this method should return IEnumerable<MgmtExtension>